src/main/java/org/example/hw05/Bank.java

```diff
@@ -0,0 +1,103 @@
+package org.example.hw05;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class Bank {
+
+    public enum SafeType {
+        UNSAFE,
+        SYNCHRONIZED,
+        REENTRANT_LOCK
+    }
+
+    private Map<Integer, Long> accounts = new HashMap<>();
+    private Random random = new Random();
+
+    public Bank(int numberOfAccounts, long minBalance, long maxBalance) {
+        for (int i = 0; i < numberOfAccounts; i++) {
+            long rndValue = random.nextLong((maxBalance - minBalance) + 1) + minBalance;
+            accounts.put(i, rndValue);
+        }
+    }
+
+    public int pickRandomAccountId() {
+        return random.nextInt(accounts.size());
+    }
+
+    public long getAccountBalance(int accountId) {
+        return accounts.get(accountId);
+    }
+
+    public void setAccountBalance(int accountId, long newBalance) {
+        accounts.put(accountId, newBalance);
+    }
+
+    public BigInteger getSumOfAllAccounts() {
+        BigInteger sum = BigInteger.ZERO;
+        for (Long value : accounts.values()) {
+            sum = sum.add(BigInteger.valueOf(value));
+        }
+        return sum;
+    }
+
+    public void transfer() {
+        int from = pickRandomAccountId();
+        int to = pickRandomAccountId();
+        long value = getAccountBalance(from);
+        long x = value == 0 ? 0 : random.nextLong(value);
+        // withdraw
+        long balFrom = getAccountBalance(from) - x;
+        setAccountBalance(from, balFrom);
+        // deposit
+        long balTo = getAccountBalance(to) + x;
+        setAccountBalance(to, balTo);
+    }
+
+    public synchronized void transferSynchronized() {
```

Comment:
it is not possible to execute more than 1 transfer at any given moment, even if accounts are different. 
please fix